event/event.go

```diff
@@ -191,15 +192,18 @@ func (ev *event) Start(ctx context.Context) error {
 		log.Warn().Str("Team 0 ", ev.store.GetTeams()[0].Name).Msg("0 indexed team....")
 	}
 	for _, team := range ev.store.GetTeams() {
-		if err := ev.AssignLab(&team); err != nil {
-			//log.Error().Msg("Error while issuing error to team.. Check out assignlab function... ")
+
+		ev.teamQueue <- team
```

Comment:
Based on what `ev.processQueue`, why introduce a channel here? As I see it, nothing is concurrent and we are effectively inserting into a channel in order to instantly read from it?

Generally, `processQueue` seems like a strange work around of fixing issues with `lab.Hub`. Unless there is something I have severely missed, I would suggest adapting the direction of the `refactor-labhub` branch instead.

Reply:
The channels are more used as a concurrently safe fifo queue, than to sync different tasks. 
Other structures could have achieved the same.

The ideas is that a team is added to the queue, and it calls the process queue.
If no teams where in the queue at that point it would just assign the team is just added, and then it's a bit stupid. 

But if there are no labs to assign it will exit early and that team will stay in the queue until the next time the `processQueue` is run.

The labhub will cal processQueue(through the hook) every time i creates a new lab thus assigning the new lab to a waiting team.

Yes the labhub design in refactor-labhub is much better, but i had the idea that this fix should be more short term.

Reply:
Ah, now it is more clear to me. The proccesQueue functions primarily as a signal for processor to perform some work while avoid blocking. The code is a little hard to grasp at first (just like the old messy version of `lab.Hub`). However, I still think the code ultimatively solve a problem with a poorly programmed `lab.Hub`. I think it is fine for this pull request. **But**, I think it would be valuable to add the refactored `lab.Hub`, as I believe it makes the code simplier (fewer lines of code), in addition to being easier to test.

Reply:
We will eventually adapt refactor-labhub 👍🏻 since it is much clear as you mentioned.